### PR TITLE
bluetooth: Implement le scan function

### PR DESCRIPTION
**Pull request by Wootak**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9ee86f127a91b60492903ca3dc20f62f71a8c02d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9ee86f127a91b60492903ca3dc20f62f71a8c02d] - Code Rule Check OK.

File: os/drivers/wireless/bluetooth/bt_null.c [hunk 1]
```diff
@@ -242,6 +256,94 @@ static void btnull_process_write_local_name_req(FAR struct bt_buf_s *buf)
...
+	uint8_t num_reports = 3;
+	uint8_t rssi = 0x9f;
+
+	nvdbg("format_le_adv_report entered.\n\n");
+
+	/* Minimal setup for the command complete event */
+
+	len = sizeof(struct bt_hci_evt_hdr_s)
+			+ sizeof(struct bt_hci_evt_le_meta_event_s)
+			+ (num_reports * (sizeof(struct bt_hci_ev_le_advertising_info_s) + sizeof(uint8_t) + 31 + 1));
```
Comment (amitpurwar) [inline os/drivers/wireless/bluetooth/bt_null.c]: I think, there is problem in len, can you please modify as below       
 len = sizeof(struct bt_hci_evt_hdr_s)
                        + sizeof(struct bt_hci_evt_le_meta_event_s)
                        + sizeof(uint8_t)
                        + (num_reports * (sizeof(struct bt_hci_ev_le_advertising_info_s) + 31 + 1));

File: os/include/tinyara/bluetooth/bt_core.h [hunk 2]
```diff
@@ -155,7 +155,7 @@ struct bt_eir_s {
 
 static inline int bt_addr_to_str(FAR const bt_addr_t *addr, FAR char *str, size_t len)
 {
-	return snprintf(str, len, "%2.2X:%2.2X:%2.2X:%2.2X:%2.2X:%2.2X", addr->val[5], addr->val[4], addr->val[3], addr->val[2], addr->val[1], addr->val[0]);
+	return snprintf(str, len, "%02X:%02X:%02X:%02X:%02X:%02X", addr->val[0], addr->val[1], addr->val[2], addr->val[3], addr->val[4], addr->val[0]);
```
Comment (amitpurwar) [inline os/include/tinyara/bluetooth/bt_core.h]: Please change addr->val[0]); to addr->val[5);

Comment (Wootak) [reply os/include/tinyara/bluetooth/bt_core.h]: Address is converted like below on capi layer. so, I changed this code sams as capi.

431 int _bt_convert_address_to_string(char **addr_str, unsigned char *addr_hex)
432 {
433     char address[18] = { 0, };
434 
435     BT_CHECK_INPUT_PARAMETER(addr_hex);
436 
437     snprintf(address, 18, "%02X:%02X:%02X:%02X:%02X:%02X", addr_hex[0], addr_hex[1], addr_hex[2], addr_hex[3], addr_hex[4], addr_    hex[5]);
438     *addr_str = strdup(address);

File: os/drivers/wireless/bluetooth/bt_null.c (hunk 1 above)
Comment (Wootak) [reply os/drivers/wireless/bluetooth/bt_null.c]: You are right. Thank you.

**Comment by seinfra** [comment]
Target : [6628c0118414df292eec5c4251259e9f9167f277] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6628c0118414df292eec5c4251259e9f9167f277] - Code Rule Check OK.

**Pull request merged**